src/com/cli/contact/manager/Searcher.java

```diff
@@ -0,0 +1,8 @@
+package com.cli.contact.manager;
+import com.cli.contact.model.Person;
+import java.util.List;
+
+public interface  Searcher {
```

Comment:
@anghelitalo Great idea handle the searching as an interface, but you have classes that have only one method you could have the interface PersonSearcher and define the 3 methods in this, so you can implement just one class with all the kind of searchers that you need